eagleproject/core/blockchain/harvest/transaction_history.py

```diff
@@ -1640,7 +1659,7 @@ def _daily_rows(steps, latest_block_number, project, start_at=0):
 
                 
             s.apr = (
-                Decimal(100) * change * (Decimal(365) * BLOCKS_PER_DAY) / blocks
+                Decimal(100) * change * Decimal(365)
```

Comment:
I don't think this change is correct. My reasoning: 
- APR is an annual percentage rate. Meaning how much % of the base value can be gained in a year's time
- in general the major loop of this functions goes through block numbers of each day in question. While the block ranges between the block numbers should be roughly a day apart... that isn't guaranteed. Especially not for the latest day
- so normalizing any earnings to earnings in a span of the day is IMO important. Before multiplying it with 365 -> which represents earnings in a year (APR)

E.g. if the block span in 1/2 a day instead of a day. And you multiply that with 365 without normalizing it the resulting APR will be 50% of the actual one.

Reply:
If the block span is less than day, wouldn't the left over yield be picked up by another day instance? Though it seems the block numbers for each day fall on 7am +/-2 mins pretty consistently. I don't think the current day is an issue atm as it doesn't show yield for a less than full 24h period.

What if the lower block count is due to longer block times rather than analytics capturing the interval too early? The yield would be accurate for a 24h period, but would still be altered by the multiplier

With each individual day's yield being manipulated, I think that makes the resulting average quite innacurate (maybe unless you get the BLOCKS_PER_DAY constant exactly right). Removing the multiplier might make the yield slightly off on a daily basis if in fact the blocks are seperated by a wrong interval, but should be accurate on a longer scale

Reply:
I see your point. 

Doesn't this function calculate the yield of the current day as well? That one will need the block normalization.

Reply:
I think it looks at the rebase event logs and uses the rebase amount to determine apy, so the current period doesn't have an event available. I can try and add that to the pr, because it seems there's an issue open for current-day yield anyway

Reply:
Maybe better to add it in a separate PR just to have thinks more separated. Cool we can keep the normalisation of events out of it then. I think it would be beneficial to add 2 things: 
- comments explaining why we are not doing any block normalisation
- a check that the 2 rebase events between which the yield is being calculated are roughly a day apart. Lets say between 23 & 25 hours. If they are outside that interval I would log a warning. So we have at least some signal that the logs come in intervals we expect.

Reply:
those are added now, thanks

Reply:
nice, thanks!